src/magicbox/SuggestionsManager.ts

```diff
@@ -28,15 +34,46 @@ enum Direction {
   Right = 'Right'
 }
 
+export interface IPopulateSearchResultPreviewsEventArgs {
+  suggestionText: string;
+  previewsQuery: Promise<ISearchResultPreview[]>;
+}
+
+export enum SuggestionsManagerEvents {
+  PopulateSearchResultPreviews = 'populateSearchResultPreviews'
+}
+
 export class SuggestionsManager {
   public hasSuggestions: boolean;
   private pendingSuggestion: Promise<Suggestion[]>;
   private options: SuggestionsManagerOptions;
-  private keyboardFocusedSuggestion: HTMLElement;
+  private keyboardFocusedElement: HTMLElement;
```

Comment:
This property was renamed for lucidity because it can refer to both a keyboard focused suggestion or a keyboard focused preview.